packages/palette/src/elements/Button/Button.tsx

```diff
@@ -172,214 +168,219 @@ export const getStylesForVariant = (variant: ButtonVariant) => {
                 }
-              `
-          }};
-        `
-      case "noOutline":
-        return css`
-          ${props => {
-            const { colors } = props.theme
-            return `
-                background-color: transparent;
-                border-color: transparent;
-                color: ${colors.black100};
-              `
-          }};
-        `
-      default:
-    }
+              }
+            `
+        }};
+      `
+    case "noOutline":
+      return css`
+        ${props => {
+          const { colors } = props.theme
+          return `
+              background-color: transparent;
+              border-color: transparent;
+              color: ${colors.black100};
+            `
+        }};
+      `
+    default:
   }
+}
 
-  render() {
-    const buttonProps = {
-      ...this.props,
-      ...this.getSize(),
-      buttonSize: this.props.size,
-      variantStyles: this.getVariant(),
-    }
-
-    return <ButtonBase {...buttonProps}>{this.props.children}</ButtonBase>
-  }
+/** A button with various size and color settings */
+export const Button: React.FC<ButtonProps> = ({
+  size = defaultSize,
+  variant = defaultVariant,
+  children,
+  ...rest
+}) => {
+  return (
+    <ButtonBase
+      {...rest}
+      {...getSize({ inline: rest.inline, size })}
+      buttonSize={size}
+      variantStyles={getVariant({ variant })}
+    >
+      {children}
+    </ButtonBase>
+  )
 }
 
 /** A base from which various button implementations can compose from */
-export class ButtonBase extends Component<ButtonBaseProps & SansProps> {
-  static defaultProps = {
-    border: 1,
-    borderRadius: 3,
-  }
-
-  onClick = event => {
-    if (!this.props.loading && this.props.onClick) {
-      this.props.onClick(event)
+export const ButtonBase: React.FC<ButtonBaseProps & SansProps> = ({
+  children,
+  loading,
+  disabled,
+  color,
+  size,
+  longestText,
+  weight,
+  className,
+  onClick,
+  ...rest
+}) => {
+  const loadingClass = loading ? "loading" : ""
+  const disabledClass = disabled ? "disabled" : ""
+  const classNames = [loadingClass, disabledClass, className]
+    .filter(Boolean)
+    .join(" ")
+
+  const handleClick = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) => {
+    if (!loading && onClick) {
+      onClick(event)
     }
   }
 
-  render() {
-    const {
-      children,
-      loading,
-      disabled,
-      color,
-      size,
-      longestText,
-      weight,
-      onClick,
-      className,
-      ...rest
-    } = this.props
-
-    const loadingClass = loading ? "loading" : ""
-    const disabledClass = disabled ? "disabled" : ""
-
-    return (
-      <Container
-        className={[loadingClass, disabledClass, className]
-          .filter(Boolean)
-          .join(" ")}
-        onClick={this.onClick}
-        disabled={disabled}
-        {...rest}
-      >
-        {loading && <Spinner size={this.props.buttonSize} />}
-
-        {longestText ? (
```

Comment:
I remember reading the PR for this one 😈

Reply:
Kieren driving his BMW around lol